single_kernel_mongo/state/unit_peer_state.py

```diff
@@ -99,3 +100,14 @@ def node_port(self) -> int:
         K8s-only.
         """
         return self.k8s.get_node_port(MongoPorts.MONGOS_PORT)
+
+    @property
+    def drained(self) -> bool:
+        """Is the shard drained."""
```

Comment:
I believe the reason that this must be a unit state instead of a peer state is because we can then check if a shard is drained *without* receiving a `relation_changed_event`. If that is true would you mind adding that context to the doc strings in these functions?

Reply:
Hum, I'm not sure what you mean by that, it was in the Unit peer databag in the former charm so I have to keep it that way I think and I don't see which difference you make between unit state and peer state in that case.
Anyway, updating our unit peer databag does not generate relation changed events IIRC and reading the drained value should not trigger a relation changed event because we are reading.
Maybe you are talking about writing the drained value in the databag?

Reply:
My apologies my suggestion wasn't clear. Let me make that clearer. Please update this comment to say:
"""
Returns True if the shard is drained.

We check the unit databag rather than the app databag since a draining operation blocks all events from occurring. The unit databag allows us to update and check our draining status in the same event as the draining. Unlike the app databag which triggers requires a `relation_changed_event` to propogate.
"""